### DISCUSSION TITLE
Map usb0 to network folder on pc

### DISCUSSION THREAD

**Discussion by efargas** [aldostools/webMAN-MOD#1061] [Q&A]
It's possible to map a network folder as usb0 to backup hdd directly to pc over network?

**Comment by aldostools** [comment]
No. /net0 is mounted as /dev_bdvd in read only mode. It would require additional hooks in Cobra and changes in netiso to handle these calls.

If I recall correctly multiMAN's file manager allows file copy to net0.

**Reply by efargas** [reply]
The idea is to use de backup utility of the system without using external drive connected to usb, its possible to do that?

**Reply by aldostools** [reply]
As I commented earlier, to write to /net0 using the file system syscalls (cellFsWrite, cellFsMkdir, cellFsDelete, etc.) requires a large modification to Cobra payload and netiso plugin which is the proxy that handles the communication between PS3 and ps3netsrv.

Cobra/PS3HEN payloads use a MITM attack intercepting all SCSI commands received by the Bluray drive from GameOS and redirect the request to different functions depending the type of storage of the mounted game or folder.

-If the source is a physical media (CD/DVD/Bluray/etc) the SCSI commands are sent directly
-If the source is an ISO stored in a native file system (HDD, USB, SD, MS, etc.) the SCSI commands are processed internally by Cobra to redirect the read of requested sectors from the physical media to the ISO file.
-If the source is a folder stored in a native file system (HDD, USB, SD, MS, etc.) the folder is mounted as /dev_bdvd using map path redirection of files. All the I/O requests are handled directly by GameOS.
-If the source of the ISO is not a native file system (NET, NTFS, EXFAT, etc.) the SCSI commands are sent to a proxy service. The proxy is an external VSH plugin (netiso, rawseciso, etc.) that handles the requests for read sectors to the ISO and returns the response to Cobra. 
-If the remote source is a folder, ps3netsrv generates a virtual ISO of the directory and the folder is handled as a read-only ISO.

As you can see for the network communication through XMB is designed to be read-only and change it to support read/write would require big changes in design and implementation of Cobra/PS3HEN and netiso plugin.

On the other hand, multiMAN's file manager implemented support for ps3netsrv API which allows read/write command. Therefore it's possible to create folders, copy files, delete files and folders, read directory, etc.

webMAN MOD also has implemented ps3netsrv API but only the read commands for internal functions like scan games or copy files from the PC to PS3.
